### PR TITLE
USWDS - Dependencies: Update prettier `3.4.2`

### LINKED ISSUE #6256: USWDS - Dependencies: Update Prettier
# Description

Prettier has released new minor versions: `3.4.x`.

After updating, Prettier has introduced changes to how it formats scss and when it breaks content onto new lines.

Originally part of POAM (#6255) but decided to break out for more confined testing.

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

Update Prettier to `3.4.2` and format Scss.

## Breaking change

This is not a breaking change.

## Related issue

Closes #6256 

## Related pull requests

#6255 

## Preview link

Preview link →

## Problem statement

Prettier has released new minor versions: 3.4.x.

After updating, Prettier has introduced changes to how it formats scss and when it breaks content onto new lines.

## Solution

Update Prettier and run formatted code.

## Testing and review

1. Inspect code changes
2. Confirm all changes are related to spacing and not changing any styles
3. Compare output CSS from `develop` and this branch using a tool like [diffchecker](https://www.diffchecker.com/)
4. Confirm CSS remains unchanged.

## Dependency updates

| Dependency name              | Previous version | New version |
| ---------------------------- | :--------------: | :---------: |
| Prettier |     3.3.3      |   3.4.2   |

### CODE REVIEW

**Review by mejiaj — COMMENTED** [review]
LGTM. Can we pull in the latest from `develop` and run again?

**Review by amyleadem — APPROVED** [review]
LGTM!
- [x] Confirm 3.4.2 is the latest version of prettier
- [x] Confirm no visual regressions on affected components
- [x] Confirm code changes are only related to spacing/formatting
- [x] Confirm I can run `npm install`, `npm start`, `npm run test` without error
- [x] Confirm no new changes when running `npm run prettier`

**Review by mejiaj — APPROVED** [review]
LGTM!

File: packages/usa-card/src/styles/_usa-card.scss [hunk 1]
```diff
@@ -255,14 +255,12 @@
     .usa-card__img {
       @include u-radius(0);
       border-top-left-radius: calc(
-        #{radius($theme-card-border-radius)} - #{units(
-            $theme-card-border-width
-          )}
+        #{radius($theme-card-border-radius)} -
+          #{units($theme-card-border-width)}
```
Comment (thisisdano) [inline packages/usa-card/src/styles/_usa-card.scss:259]: I like this change, more readable!

File: packages/usa-step-indicator/src/styles/_usa-step-indicator.scss [hunk 2]
```diff
@@ -203,10 +202,10 @@ $step-indicator-counter-size-sm: 3;
...
         top: calc(
           (
-              #{units($step-indicator-counter-size)} - #{units(
-                  $theme-step-indicator-segment-height
-                )}
-            ) / -2
+              #{units($step-indicator-counter-size)} -
+                #{units($theme-step-indicator-segment-height)}
+            ) /
+            -2
```
Comment (thisisdano) [inline packages/usa-step-indicator/src/styles/_usa-step-indicator.scss:208]: This change feels less intuitive, but 🤷‍♀️

**Review by thisisdano — APPROVED** [review]
Good legibility improvements throughout

**Pull request merged**